rasa_core/train.py

```diff
@@ -189,82 +177,171 @@ def train_dialogue_model(domain_file, stories_file, output_path,
 
-    additional_arguments = _additional_arguments(cmdline_args)
-
     if cmdline_args.url:
         stories = utils.download_file_from_url(cmdline_args.url)
     else:
         stories = cmdline_args.stories
 
-    _endpoints = AvailableEndpoints.read_endpoints(cmdline_args.endpoints)
-    _interpreter = NaturalLanguageInterpreter.create(cmdline_args.nlu,
-                                                     _endpoints.nlu)
-    _broker = PikaProducer.from_endpoint_config(_endpoints.event_broker)
-    _tracker_store = TrackerStore.find_tracker_store(None,
-                                                     _endpoints.tracker_store,
-                                                     _broker)
-    if cmdline_args.core:
-        if not cmdline_args.interactive:
-            raise ValueError("--core can only be used together with the"
-                             "--interactive flag.")
-        elif cmdline_args.finetune:
-            raise ValueError("--core can only be used together with the"
-                             "--interactive flag and without --finetune flag.")
-        else:
-            logger.info("loading a pre-trained model. ",
-                        "all training-related parameters will be ignored")
-        _agent = Agent.load(cmdline_args.core,
-                            interpreter=_interpreter,
-                            generator=_endpoints.nlg,
-                            tracker_store=_tracker_store,
-                            action_endpoint=_endpoints.action)
-    else:
+    if cmdline_args.mode == 'default':
         if not cmdline_args.out:
             raise ValueError("you must provide a path where the model "
                              "will be saved using -o / --out")
-        _agent = train_dialogue_model(cmdline_args.domain,
-                                      stories,
-                                      cmdline_args.out,
-                                      _interpreter,
-                                      _endpoints,
-                                      cmdline_args.history,
-                                      cmdline_args.dump_stories,
-                                      cmdline_args.config,
-                                      additional_arguments)
-
-    if cmdline_args.interactive:
+        if (isinstance(cmdline_args.config, list) and
+                len(cmdline_args.config) > 1):
+            raise ValueError("You can only pass one config file at a time")
+
+        _agent = train_dialogue_model(domain_file=cmdline_args.domain,
+                                      stories_file=stories,
+                                      output_path=cmdline_args.out,
+                                      dump_stories=cmdline_args.dump_stories,
+                                      policy_config=cmdline_args.config[0],
+                                      kwargs=additional_arguments)
+
+    elif cmdline_args.mode == 'interactive':
```

Comment:
can we put the different modes into separate functions to not put to much code directly into `if __name__ = "main"`